### PR TITLE
Add monitor state to monitor results

### PR DESCRIPTION
**Pull request by Seanstoppable**
This allows one to use the api to trigger actions based on a monitor in
'Alert' status, as an example

### CODE REVIEW

File: monitors.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ type Monitor struct {
 	Query   *string  `json:"query,omitempty"`
 	Name    *string  `json:"name,omitempty"`
 	Message *string  `json:"message,omitempty"`
+	State   *string  `json:"overall_state,omitempty"`
```
Comment (ojongerius) [inline monitors.go]: When checking the DD API docs at https://docs.datadoghq.com/api/?lang=bash#monitors I noticed both "state" and "overal_state" exist, and have different meanings. Unmarshalling "overall_state" into "State" would be confusing.

**Comment by ojongerius** [comment]
Thanks for taking the time to create this PR @Seanstoppable!

When reading up on the API doco at https://docs.datadoghq.com/api/?lang=bash#monitors I noticed there are quite a few changes possible in regards to states:

## Input
* `group_states`: could be "all" or "alert,warn" used as a an argument to getting monitor details

## Output
* `state`: empty or contains an array of `groups`, consisting of `hosts`
* `overall_state`: `*string`, example "Alert" -as seen in example of "resolve monitor"
* `overall_state_modified`:  `*string`: "2017-08-11T12:21:05.949182+00:00" -as seen in example of "resolve monitor"

I'be be happy to merge this afer you address my inline comment. However it would be wonderful if you had the time to implement `state, `overall_state` and `overal_state_modified`.

File: monitors.go (hunk 1 above)
Comment (Seanstoppable) [reply monitors.go]: Completely missed regular state, thanks! I've updated this to be inline with the json.

**Comment by Seanstoppable** [comment]
I'm happy to look at further expansion in my spare time. `OverallState` is just something I am trying to access for another project.

**Comment by ojongerius** [comment]
@Seanstoppable great! I've raised #164 feel free to call dibs on it.

**Pull request merged**